darshan-util/pydarshan/tests/test_plot_exp_common.py

```diff
@@ -0,0 +1,172 @@
+            plot_access_histogram,
+            [2289, 1193, 2470, 173, 1, 0, 0, 0, 0,
+            0, 160, 12, 1325, 0, 0, 0, 0, 0, 0, 0],
+        ),
+        (
+            "examples/example-logs/ior_hdf5_example.darshan",
+            "POSIX",
+            plot_access_histogram,
+            [3, 17, 0, 0, 16, 0, 0, 0, 0, 0, 3, 4, 0, 0, 16, 0, 0, 0, 0, 0],
+        ),
+        (
+            "examples/example-logs/ior_hdf5_example.darshan",
+            "MPI-IO",
+            plot_access_histogram,
+            [3, 17, 0, 0, 16, 0, 0, 0, 0, 0, 3, 4, 0, 0, 16, 0, 0, 0, 0, 0],
+        ),
+        (
+            "examples/example-logs/sample-badost.darshan",
+            "POSIX",
+            plot_access_histogram,
+            [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 131072, 0, 0, 0, 0],
+        ),
+        # subtle case where "POSIX" and "MPIIO"
+        # modules show different results
+        (
+            "tests/input/sample-dxt-simple.darshan",
+            "POSIX",
+            plot_access_histogram,
+            [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 1, 0, 0, 0, 0, 0, 0, 0],
+        ),
+        (
+            "tests/input/sample-dxt-simple.darshan",
+            "MPI-IO",
+            plot_access_histogram,
+            [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0],
+        ),
+        # more obvious case where "POSIX" and
+        # "MPIIO" modules show different results
+        (
+            None,
+            "POSIX",
+            plot_access_histogram,
+            [18, 2492, 14679, 0, 50486, 186, 0, 0, 0,
+            0, 43, 301, 2, 0, 50486, 0, 0, 0, 0, 0],
+        ),
+        (
+            None,
+            "MPI-IO",
+            plot_access_histogram,
+            [11, 2492, 2, 0, 0, 0, 0, 410, 86, 0, 2526,
+            303, 2, 0, 97812, 396, 0, 410, 86, 0],
+        ),
+        (
+            "examples/example-logs/dxt.darshan",
+            None,
+            plot_opcounts,
+            [6126, 1497, 264, 1379, 5597, 0, 0, 0, 0, 0, 0, 0,
+            0, 0, 0, 0, 0, 0, 0, 0, 0, 39, 0, 1, 0, 0, 0, 0],
+        ),
+        (
+            "examples/example-logs/ior_hdf5_example.darshan",
+            None,
+            plot_opcounts,
+            [36, 23, 22, 4, 53, 0, 0, 36, 23, 1, 0, 0, 0,
+            0, 0, 0, 16, 0, 0, 0, 0, 0, 128, 1, 0, 0, 0, 9],
+        ),
+        (
+            "examples/example-logs/sample-badost.darshan",
+            None,
+            plot_opcounts,
+            [0, 131072, 2048, 2048, 131072, 0, 2048, 0, 0, 0, 0, 0, 0,
+            0, 0, 0, 0, 0, 0, 0, 0, 34816, 97, 6144, 0, 0, 0, 2056],
+        ),
+        (
+            "examples/example-logs/sample.darshan",
+            None,
+            plot_opcounts,
+            [0, 16402, 2049, 0, 16404, 0, 0, 0, 18, 0, 0, 0, 0, 0,
+            0, 16384, 2048, 0, 0, 0, 0, 0, 74, 129, 0, 0, 0, 0],
+        ),
+    ],
+)
+@pytest.mark.skipif(not pytest.has_log_repo, # type: ignore
+                    reason="missing darshan_logs")
```

Comment:
Why skip every single test case in `test_bar_heights()` when the logs repo is absent? I think the design we'd want is to only skip the cases with filepaths we don't have access to, but the in-repo files we currently always have access to.

`pytest` provides facilities for more fine-grained skipping, and also for imperative skipping in the test proper when that is more convenient.

Reply:
Yeah this was a bit too coarse. I changed to marking the individual test cases to skip which seems much more appropriate. Locally it works for me with/without the logs repo available.